### DISCUSSION TITLE
How can I build qt5-base with vendor dependecies?

### DISCUSSION THREAD

**Discussion by tersion** [microsoft/vcpkg#24164] [General]
I need to build **qt5-base** with some of vendor (built in 3rd party libs) dependencies like **freetype**, **harfbuzz** and so on.
I modified **portfile.cmake**, and commented out this piece:
```
# Remove vendored dependencies to ensure they are not picked up by the build
#foreach(DEPENDENCY zlib freetype harfbuzz-ng libjpeg libpng double-conversion sqlite pcre2)
#    if(EXISTS ${SOURCE_PATH}/src/3rdparty/${DEPENDENCY})
#        file(REMOVE_RECURSE ${SOURCE_PATH}/src/3rdparty/${DEPENDENCY})
#    endif()
#endforeach()
```
Then I also commented lines like, which search path to specific lib which I don't need and want use built in **qt5-base** itself:
`find_library(FREETYPE_RELEASE NAMES freetype PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH)`
Then in vcpkg.json of **qt5-base** port I remove corespondent dependencies.
And when I try to install through `vcpkg install qt5-base:x86-windows-static`, I get errors like:
```
Checking for HarfBuzz... 
Trying source 0 (type inline) of library harfbuzz ...
harfbuzz/hb.h not found in [] and global paths.
  => source produced no result.
test config.qtbase_gui.libraries.harfbuzz FAILED
```
And:
```
ERROR: Feature 'system-freetype' was enabled, but the pre-condition 'features.freetype && libs.freetype' failed.
ERROR: Feature 'system-harfbuzz' was enabled, but the pre-condition 'features.harfbuzz && libs.harfbuzz' failed.
ERROR: Feature 'system-png' was enabled, but the pre-condition 'features.png && libs.libpng' failed.
```
Here the link for full error log: https://pastebin.com/1R73YRSK
To build, I use:
OS: Windows 10 Pro x64
IDE: VS 2019
So, how can I build qt5-base with built in libs like: zlib, freetype, harfbuzz-ng, libjpeg, libpng, double-conversion, sqlite, pcre2, without using vcpkg ports?

**Comment by Osyotr** [comment]
Try removing `-system-whatever` from here: https://github.com/microsoft/vcpkg/blob/05bcc5fca4ec1bd11a1e1a655ff37752dc5737f8/ports/qt5-base/portfile.cmake#L120
